commit c465def6bfe834b62623caa9b98f2d4f4739875a
Author: Ying Huang <huang.ying.caritas@gmail.com>
Date:   Mon Jun 15 10:42:57 2009 +0800

    PCI AER: software error injection
    
    Debugging PCIE AER code can be very difficult because it is hard
    to trigger various real hardware errors. This patch provide a
    software based error injection tool, which can fake various PCIE
    errors with a user space helper tool named "aer-inject". Which
    can be gotten from:
    
      http://www.kernel.org/pub/linux/kernel/people/yhuang/
    
    The patch fakes AER error by faking some PCIE AER related
    registers and an AER interrupt for specified the PCIE device.
    
    Signed-off-by: Huang Ying <ying.huang@intel.com>
    Signed-off-by: Jesse Barnes <jbarnes@virtuousgeek.org>

diff --git a/drivers/pci/pcie/aer/Kconfig.debug b/drivers/pci/pcie/aer/Kconfig.debug
new file mode 100644
index 000000000000..b8c925c1f6aa
--- /dev/null
+++ b/drivers/pci/pcie/aer/Kconfig.debug
@@ -0,0 +1,18 @@
+#
+# PCI Express Root Port Device AER Debug Configuration
+#
+
+config PCIEAER_INJECT
+	tristate "PCIE AER error injector support"
+	depends on PCIEAER
+	default n
+	help
+	  This enables PCI Express Root Port Advanced Error Reporting
+	  (AER) software error injector.
+
+	  Debuging PCIE AER code is quite difficult because it is hard
+	  to trigger various real hardware errors. Software based
+	  error injection can fake almost all kinds of errors with the
+	  help of a user space helper tool aer-inject, which can be
+	  gotten from:
+	     http://www.kernel.org/pub/linux/utils/pci/aer-inject/